src/main/java/net/neoforged/neoforge/client/event/RegisterJsonAnimationTypesEvent.java

```diff
@@ -33,13 +39,26 @@ public RegisterJsonAnimationTypesEvent(
      * Register a custom {@link AnimationTarget} with the specified {@code key}.
      */
     public void registerTarget(ResourceLocation key, AnimationTarget target) {
+        checkDuplicate("target", key, targetsRegisteredBy);
         targets.put(key, target);
     }
 
     /**
      * Register a custom {@link AnimationChannel.Interpolation interpolation function} with the specified {@code key}.
      */
     public void registerInterpolation(ResourceLocation key, AnimationChannel.Interpolation interpolation) {
+        checkDuplicate("interpolation", key, interpolationsRegisteredBy);
         interpolations.put(key, interpolation);
     }
+
+    private static void checkDuplicate(String what, ResourceLocation key, Map<ResourceLocation, String> by) {
+        final var currentModId = ModLoadingContext.get().getActiveContainer().getModId();
```

Comment:
Use of the magic thread-local mod container is being phased out. The duplicate check should be as simple as this:
```java
AnimationTarget oldTarget = targets.put(key, target);
if (oldTarget != null) {
    throw new IllegalStateException(String.formatted(
            Locale.ROOT,
            "Duplicate target registration for key '%s' (old: %s, new: %s)",
            key,
            oldTarget,
            target));
}
```